src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -147,15 +157,27 @@ protected override void ApplyResponseChallenge()
                 properties = new AuthenticationProperties(ChallengeContext.Properties);
             }
 
-            if (string.IsNullOrEmpty(properties.RedirectUri))
+            string redirectToUse = null;
```

Comment:
We should discuss this if it is not clear. One way that properties.RedirectUri gets set is when a user issues a challenge, as opposed to hitting the [Authorize] attribute. In that case, users have expressed the need and desire to be able to set the redirect_uri from the challenge overriding the value on Options. If you think we should distinguish between the two then we should perhaps introduce a new value to capture this intent and keep the two separate.